js/shared/icecastMetadata.js

```diff
@@ -0,0 +1,176 @@
+                    var stream = streams[i];
+                    var streamUrl = stream.SERVERURL[0];
+                    if (streamUrl == url) {
+                        stationStats = stream;
+                    }
+                }
+            }
+            if (stationStats != null && stationStats.SONGTITLE) {
+                var station = {};
+                station.listeners = stationStats.CURRENTLISTENERS[0];
+                station.bitrate = stationStats.BITRATE[0];
+                station.title = stationStats.SONGTITLE[0];
+                station.fetchsource = 'SHOUTCAST_V2';
+                callback(station);
+            }
+        });
+    }
+
+    function getIcecastStation(url, callback) {
+        var parser = document.createElement('a');
+        parser.href = url;
+        var furl = parser.protocol + '://' + parser.hostname + ':' + parser.port + '/status-json.xsl';
+        fetch(furl, {
+            method: 'GET'
+        })
+        .then(response => response.json())
+        .then(body => {
+            if (
+                !body.icestats ||
+                !body.icestats.source ||
+                body.icestats.source.length === 0
+            ) {
+                return;
+            }
+            var sources = body.icestats.source;
+            for (var i = 0, mountCount = sources.length; i < mountCount; i++) {
+                var source = sources[i];
+                if (source.listenurl === url) {
+                    var station = {};
+                    station.listeners = source.listeners;
+                    station.bitrate = source.bitrate;
+                    station.title = source.title;
+                    station.fetchsource = 'ICECAST';
+                    callback(station);
+                }
+            }
+        });
+    }
+
+    function getStreamStation(url, callback) {
+        let controller = new AbortController();
+        const options = {
+            metadataTypes: ['icy'],
+            icyCharacterEncoding: 'utf-8',
+            icyMetaInt: 16000,
+            icyDetectionTimeout: 2000,
+            enableLogging: false,
+            onStream: () => {},
+            onMetadata: (value) => {var station = {};
+                                   station.title = value.metadata.StreamTitle;
+                                   station.fetchsource = 'STREAM';
+                                   callback(station);
+                                   controller.abort();},
+            onError: () => {}
+        };
+
+        fetch(url, {
+            method: 'GET',
+            headers: {
+                'Icy-MetaData': '1',
+            },
+            signal: controller.signal
+        })
+        .then(async (response) => {
+            const icecast = new IcecastReadableStream(
+                response,
+                options
+            );
+
+            await icecast.startReading();
+        });
+    }
+
+    var methodHandler = undefined;
+
+    switch (method) {
```

Comment:
Apparently the code is now always using the "STREAM" method? Is there any reason to have these others here? In which case would they be useful?

Reply:
Yes, I was preparing add support to for shoutcast stream title using url parsing, most of radios today use icecast metadata so is not important.